app/controllers/chargeback_controller.rb

```diff
@@ -390,47 +390,39 @@ def cb_rpts_fetch_saved_report(id)
       return
     end
     @right_cell_text ||= _("Saved Chargeback Report [%{name}]") % {:name => rr.name}
-    if rr.userid != session[:userid]
+    if !current_user.miq_group_ids.include?(rr.miq_group_id) && !admin_user?
       add_flash(_("Report is not authorized for the logged in user"), :error)
       @saved_reports = cb_rpts_get_all_reps(id.split('-')[1])
       return
     else
       @report_result_id = session[:report_result_id] = rr.id
       session[:report_result_runtime]  = rr.last_run_on
-      @report = rr.report_results
-      session[:rpt_task_id] = nil
-      if @report.blank?
-        add_flash(_("Saved Report \"%{name}\" not found, Schedule may have failed") %
-          {:name => format_timezone(rr.last_run_on, Time.zone, "gtl")}, :error)
-        @saved_reports = cb_rpts_get_all_reps(rr.miq_report_id.to_s)
-        rep = MiqReport.find_by_id(rr.miq_report_id)
-        if x_active_tree == :cb_reports
-          self.x_node = "reports-#{rep.id}"
+      if rr.status.casecmp("finished") == 0
+        @report = rr.report_results
+        session[:rpt_task_id] = nil
+        if @report.blank?
+          add_flash(_("Saved Report \"%{name}\" not found, Schedule may have failed") %
+            {:name => format_timezone(rr.last_run_on, Time.zone, "gtl")}, :error)
+          @saved_reports = cb_rpts_get_all_reps(rr.miq_report_id.to_s)
+          rep = MiqReport.find_by_id(rr.miq_report_id)
+          if x_active_tree == :cb_reports_tree
+            self.x_node = "reports-#{rep.id}"
+          end
+          return
         else
-          @sb[:rpt_menu].each_with_index do |lvl1, i|
```

Comment:
@lpichler It's not obvious to me why this is being removed. Can you provide some details? Thanks!